metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -355,26 +368,37 @@ private void periodicCheckingLoop(long sleepTime, AbstractExecutablePlugin plugi
         currentThread().interrupt();
         return;
       } catch (ExternalTaskException e) {
-        LOGGER.warn(String
-            .format("workflowExecutionId: %s, pluginType: %s - ExternalTaskException occurred.",
-                workflowExecution.getId(), plugin.getPluginType()), e);
-        if (!ExternalRequestUtil.doesExceptionCauseMatchAnyOfProvidedExceptions(
-            UNMODIFIABLE_MAP_WITH_NETWORK_EXCEPTIONS, e)) {
-          // Set plugin to FAILED and return immediately
-          plugin.setFinishedDate(null);
-          plugin.setPluginStatusAndResetFailMessage(PluginStatus.FAILED);
-          plugin.setFailMessage(String.format(DETAILED_EXCEPTION_FORMAT, MONITOR_ERROR_PREFIX,
-              ExceptionUtils.getStackTrace(e)));
+        final Throwable cause = ExternalRequestUtil.getCause(e);
+        if (cause instanceof IllegalStateException) {
+          //If the application has a forcible shutdown we might experience the JerseyClient in DpsClient
+          // internally closing down without our consent even if we would have a synchronized close
+          // implemented. This catch gives us a little more assurance of avoiding an execution
+          // being marked as failed.
+          LOGGER.debug("Application is probably shutting down at the moment and dpsClient has "
+              + "closed without our consent, so we are ignoring this exception.", e);
           return;
-        }
-        consecutiveCancelOrMonitorFailures++;
-        LOGGER.warn(String.format(
-            "workflowExecutionId: %s, pluginType: %s - Monitoring of external task failed %s "
-                + "consecutive times. After exceeding %s retries, pending status will be set",
-            workflowExecution.getId(), plugin.getPluginType(), consecutiveCancelOrMonitorFailures,
-            MAX_CANCEL_OR_MONITOR_FAILURES), e);
-        if (consecutiveCancelOrMonitorFailures >= MAX_CANCEL_OR_MONITOR_FAILURES) {
-          plugin.setPluginStatusAndResetFailMessage(PluginStatus.PENDING);
+        } else {
```

Comment:
Else is not needed here - the if has a return at the end.

Reply:
Fixed